platforms/core-configuration/file-collections/src/integTest/groovy/org/gradle/api/file/FileCollectionLifecycleIntegrationTest.groovy

```diff
@@ -21,6 +21,79 @@ import org.gradle.api.tasks.TasksWithInputsAndOutputs
 import org.gradle.integtests.fixtures.AbstractIntegrationSpec
 
 class FileCollectionLifecycleIntegrationTest extends AbstractIntegrationSpec implements TasksWithInputsAndOutputs {
+    def "UPGRADED task #annotation configurable file collection is NOT implicitly finalized when task starts execution"() {
+        executer.requireOwnGradleUserHomeDir("temp")
+        buildFile << """
+            import org.gradle.internal.instrumentation.api.annotations.ReplacesEagerProperty
+
+            abstract class SomeTask extends DefaultTask {
+                @ReplacesEagerProperty
+                ${annotation}
+                abstract ConfigurableFileCollection getProp()
+
+                @TaskAction
+                void go() {
+                    println "value: " + prop.files
+                }
+            }
+
+            task show(type: SomeTask) {
+                def layout = project.layout
+                prop.setFrom([layout.projectDir.file("in.txt")])
+                doFirst {
+                    prop.setFrom([layout.projectDir.file("other.txt")])
+                }
+            }
+        """
+
+        expect:
+        executer.expectDeprecationWarningWithPattern("Changing property value of task ':show' property 'prop' at execution time. This behavior has been deprecated.*")
+        succeeds("show")
```

Comment:
❓ Should we also assert that the task action shows the expected value? (I assume it is the value from `doFirst`). This also applies to other tests.

Reply:
Makes sense, yes